js/modules/k6/html/html.go

```diff
@@ -373,20 +373,28 @@ func (s Selection) Is(v goja.Value) bool {
 	}
 }
 
-func (s Selection) Map(v goja.Value) (result []string) {
+func (s Selection) Map(v goja.Value) []string {
 	gojaFn, isFn := goja.AssertFunction(v)
 	if !isFn {
 		common.Throw(s.rt, errors.New("Argument to map() must be a function"))
 	}
 
-	fn := func(idx int, sel *goquery.Selection) string {
+	fn := func(idx int, sel *Selection) string {
 		if fnRes, fnErr := gojaFn(v, s.rt.ToValue(idx), s.rt.ToValue(sel)); fnErr == nil {
```

Comment:
Why didn't you just changed the `s.rt.ToValue(sel)` to be `s.rt.ToValue(&Selection{rt: s.rt, sel: sel, URL: s.URL}))`

Reply:
Not so certain about the URL, but everything else should work as now :) without reimplementing (albeit a simple) function from goquery

Reply:
Ah right, let me try.